### ISSUE TITLE
directory doesn't expand

### ISSUE THREAD

**Issue by muratkarakus7** [scrooloose/nerdtree#521]
Error detected while processing function nerdtree#ui_glue#invokeKeyMap..73..114..231..238:
line    3:
E117: Unknown function: strchars
E15: Invalid expression: strchars(a:line[0:numLeadBytes-1])

**Comment by oohcoder** [comment]
I also have got this error

**Comment by muratkarakus7** [comment]
My vim version is 7.0x, I am going to upgrade it. I guess it is because of
it.

---

**Comment by olehb** [comment]
When folder A contains only folder B (nothing more), and folder B contains some files, there are issues expanding folder A. Basically when hitting Enter on folder A, it doesn't expand. It shows A/B, and arrow down, but doesn't show any files.

**Comment by tolinwei** [comment]
I am having the same issue on Linux, but not on OS X.

**Comment by RaiseFree** [comment]
me too, It wouldn`t expand on OS X.

**Comment by singhan-chen** [comment]
I can reproduce this issue.

My VIM version: version 7.2.148.
NerdTree in 4ebbb533c3faf2c480211db2b547972bb3b60f2b

----------------------

The issue doesn't occur When I roll back to NerdTree 4.2.0

**Comment by aoak** [comment]
Same as @singhan-chen. Is there a workaround available? (expect rolling back to 4.2.0)

**Comment by tolinwei** [comment]
come back after receiving email notification.
@aoak @RaiseFree I fixed this by upgrading Vim to 7.4, not sure if this works for you, but i believe it's worth to have a try.

**Comment by aoak** [comment]
Yes, upgrading to vim 7.4 fixes it. Looks like vim 7.2 doesn't have ```strchars()``` function. At least it was not shown in ```:help functions```

**Comment by jmgarciamaleno** [comment]
Exactly the same behaviour as explained by @olehb but it doesn't happens until some open/close operations are done in the tree, not sure which. Vim 7.4 in Ubuntu 15.10.

**Comment by lebster** [comment]
Im having the same issue, changing the root to the directory will get it to show some files/folders but it is very inconsistent

**Comment by sucotronic** [comment]
I'm forced to use vim 7.2 in a project, and solved this using a simple subsitution seen in https://github.com/kozy4324/dotfiles/blob/master/.vim/autoload/vital/_c980ec/data/string.vim#L62

`
 function! s:UI._indentLevelFor(line) '\M\[^ '.g:NERDTreeDirArrowExpandable.g:NERDTreeDirArrowCollapsible.']')  
     "have to do this work around because match() returns bytes, not chars                                      
     let numLeadBytes = match(a:line, '\M\[^ '.g:NERDTreeDirArrowExpandable.g:NERDTreeDirArrowCollapsible.']')  
     "let leadChars = strchars(a:line[0:numLeadBytes-1])                                                        
     let leadChars = strlen(substitute(copy(a:line[0:numLeadBytes-1]), '.', 'x', 'g'))                          
                                                                                                                
     return leadChars / s:UI.IndentWid()                                                                        
 endfunction                                                                                                    
`

**Comment by nettrino** [comment]
I can also reproduce this in vVm version 7.4.576

**Comment by robert-blankenship** [comment]
@sucotronic Much appreciated! It looks like `strchars` is not available in the VIM API in certain version of VIM. 

@nettrino Interesting. It appears to work on `7.4.944` (installed with Homebrew on OSX)

We should fix this in master. I'm having the same issue when using VIM provided by RHEL package `vim-enhanced.x86_64 2:7.2.411-1.8.el6 @base` (which is basically VIM 7.2 installed with everything).

**Comment by phelpaustin** [comment]
Any updates on the fix to this defect??

**Comment by anticpp** [comment]
problem solved by upgrading vim to 7.4, i guess there is no 'strchars' in vim 7.2

**Comment by forevernull** [comment]
Vim 7.3 below has not built-in function 'strchars'`:help eval`. So we can't use nerdtree(tag 4.2.0 above).It works for me : 
```
cd  ~/.vim/bundle/nerdtree \
git checkout -b tag4.2.0 4.2.0
```

**Comment by suqingdong** [comment]
@forevernull Good! This works for me too. Mine is Vim7.2 on Linux.

**Comment by PhilRunninger** [comment]
This is fixed in commit [4428aba](https://github.com/scrooloose/nerdtree/commit/4428aba2a299cbd117c80e7d18068b2cbb222fff) using @yous 's posted workaround in #517.
